packages/ag-charts-community/src/chart/dom/domManager.ts

```diff
@@ -241,10 +255,51 @@ export class DOMManager extends BaseManager<Events['type'], Events> {
         });
     }
 
+    /** Get the main chart area client bound rect. */
     getBoundingClientRect() {
         return this.rootElements['canvas'].element.getBoundingClientRect();
     }
 
+    /**
+     * Get the client bounding rect for overlay elements that might float outside the bounds of the
+     * main chart area.
+     */
+    getOverlayClientRect() {
+        const window = getWindow();
+        const windowBBox = new BBox(0, 0, window.innerWidth, window.innerHeight);
+        const container = this.getRawOverlayClientRect();
+
+        if (container == null) return windowBBox.toDOMRect();
+
+        const containerBBox = BBox.fromDOMRect(container);
+        return windowBBox.intersection(containerBBox)?.toDOMRect() ?? NULL_DOMRECT;
+    }
+
+    private getRawOverlayClientRect() {
+        let element: HTMLElement | null = this.element;
+
+        // Try and find a parent which will clip rendering of children - if found we should restrict
+        // to that elements bounding box.
+        while (element != null) {
+            const styleMap = element.computedStyleMap?.();
```

Comment:
This isn't supported by Safari <16.4 and Firefox

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle

Reply:
I've improved the fallback handling to switch back to the canvas bounds if we can't use this.